clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/PlanningTask.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.venice.datarecovery;
+
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.controllerapi.StoreHealthAuditResponse;
+import com.linkedin.venice.meta.RegionPushDetails;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+
+
+/**
+ * A Task is wrapper class that designed to execute multiple commands in sequence.
+ */
+public class PlanningTask implements Runnable {
+  private final TaskParams taskParams;
+  private Integer estimatedTimeResult = -2;
+  private ControllerClient controllerClient;
+
+  public PlanningTask(TaskParams params, ControllerClient controllerClient) {
+    this.taskParams = params;
+    this.controllerClient = controllerClient;
+  }
+
+  @Override
+  public void run() {
+    estimatedTimeResult = 0;
+    // get store's push + partition info
+    StoreHealthAuditResponse storeHealthInfo = controllerClient.listStorePushInfo(taskParams.getStoreName(), true);
+    Map<String, RegionPushDetails> pushDetails = storeHealthInfo.getRegionPushDetails();
+    int ret = 0;
+
+    // examine and record/avg push times
+    for (Map.Entry<String, RegionPushDetails> entry: pushDetails.entrySet()) {
+      Instant startTime = Instant.parse(entry.getValue().getPushStartTimestamp() + "Z");
+      Instant endTime = Instant.parse(entry.getValue().getPushEndTimestamp() + "Z");
+      ret += startTime.until(endTime, ChronoUnit.SECONDS);
```

Comment:
Correct me if I am wrong, here we add the push time of the same store version but for different regions and use it for the estimated time. However, in the real life, those are run in parallel, so the addition seems incorrect. Wondering if it make sense to do the following:
1. Add a new arg `DEST_FABRIC` to the `ESTIMATE_DATA_RECOVERY_TIME` to indicate in which region we want to estimate the recovery time.
2. Since `listStorePushInfo` reports all region's recovery time for the input store, we can report the one corresponding to the `DEST_FABRIC`.
3. If there is a need in future, we can report some statistics of the same push job but in other fabrics, e.g. avg/max/min.

Reply:
Now checks region push detail map for the entry associated with the passed destination fabric argument, and returns the difference between those push start and end timestamps.